### PR TITLE
upddate cropduster, add tests and add getMultipleTargets

### PR DESCRIPTION
**Pull request by aqmnguyen**
## Current Behavior
At the moment there is no way to for us to send a `POST` request to sorcerer to get multiple rows back. 

## Why do we need this change?
This adds in a new method called `getMultipleTargets` which enables the user to send a `POST` request to sorcerer. 
This `POST` request will need a `body` that contains the segments for that DataSource. 
This update to sorcerer also allows for multiple targets to be passed in the `body` to get back rows that match all those targets. 

For example if you have a file like this: 

Level | Tier | Content
-- | -- | --
1 | Silver | Marry Poppins
2 | Gold | Peter Pan
3 | Platinum | Robin Hood
1 | Silver | Tom and Jerry
5 | Gold | Aladdin
1 | Gold | The Lion King
1 |   | Cinderella

You can pass a body like this: 

```js
[
  {
   Level: 1,
   Tier: 'Silver',
 },
 {
   Level: 2,
   Tier: 'Gold',
 },
]
```

Which returns the following: 

```js
[
  { Level: '1', Tier: 'Silver', Content: 'Tom and Jerry' },
  { Level: '2', Tier: 'Gold', Content: 'Peter Pan' },
  { Level: '1', Tier: 'Silver', Content: 'Marry Poppins' },
]
```

## Implementation Details

Adds `getMultipleTargets` method as a wrapper for Solutions / MCS to use in custom projects & apps. 

#### Dependencies (if any)
Upgraded `cropduster` from `5.2.0` to `7.1.0` which allows for native `async/await` rather than us having to wrap it around promises

:house: [ch43475](https://app.clubhouse.io/movableink/story/43475)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,8 +203,107 @@ When geolocating, if both targeting keys and geolocation columns are set, we wil
...
+    },
+    {
+      Level: 2,
+      Tier: 'Gold',
+    },
+  ]),
+};
+
+const source = new DataSource('some_key');
+const data = await source.getMultipleRows(options);
```
Comment (MansurTsutiev) [inline README.md:242]: looks like you named the actual method `getMultipleTargets`?

File: src/index.ts [hunk 2]
```diff
@@ -30,19 +30,42 @@ export default class DataSource {
...
+    const params = {
+      mi_multiple: true,
+      mi_include_headers: true,
+    };
+
+    const { method = null } = opts;
+
+    if (!method && method.toLowerCase() !== 'post') {
+      throw new Error('Request method must be POST for getMultipleTargets');
+    }
```
Comment (MansurTsutiev) [inline src/index.ts:66]: this works, just curious why not hard code `method: "post"` as part of the options?

**Review by MansurTsutiev — COMMENTED** [review]
Looks good 👍 , just a few non-blocking comments.

File: README.md (hunk 1 above)
Comment (aqmnguyen) [reply README.md:242]: Ahhh yes, bad copy & pasta

File: src/index.ts (hunk 2 above)
Comment (aqmnguyen) [reply src/index.ts:66]: I was thinking this, my line of thought was to make it identical to how Solutions currently makes a `CD.getCORS` request when they pass through the whole options object. 

That way if they 'forgot' to do this. I'm fine to make it hardcoded let me know what you all prefer.

Comment (MansurTsutiev) [reply src/index.ts:66]: Makes sense to keep it the same way for consistency.

**Review by MansurTsutiev — APPROVED** [review]

**Review by shyshy — APPROVED** [review]

**Pull request merged**